cypress/e2e/swap.test.ts

```diff
@@ -112,6 +115,53 @@ describe('Swap', () => {
       })
   })
 
+  it('should render an error when a transaction fails due to a passed deadline', () => {
+    const BALANCE_INCREMENT = 1
+    const DEADLINE_MINUTES = 30
+    cy.visit('/swap', { ethereum: 'hardhat' })
+      .then((window) => {
+        hardhat = window.hardhat
+      })
+      .then(() => hardhat.utils.getBalance(hardhat.wallet.address, USDC_MAINNET))
+      .then((balance) => Number(balance.toFixed(1)))
+      .then((initialBalance) => {
+        // input swap info
+        cy.get('#swap-currency-output .open-currency-select-button').click()
+        cy.get(getTestSelector('token-search-input')).clear().type(MAINNET_USDC_ADDRESS)
+        cy.contains('USDC').click()
+        cy.get('#swap-currency-output .token-amount-input').clear().type(BALANCE_INCREMENT.toString())
+        cy.get('#swap-currency-input .token-amount-input').should('not.equal', '')
+
+        // set deadline to minimum (1 minute)
+        cy.get(getTestSelector('open-settings-dialog-button')).click()
+        cy.get(getTestSelector('deadline-input')).clear().type(DEADLINE_MINUTES.toString())
+        cy.get('body').click('topRight')
+        cy.get(getTestSelector('deadline-input')).should('not.exist')
+
+        cy.get('#swap-button').click()
+        cy.get('#confirm-swap-or-send').click()
+
+        cy.then(() => hardhat.send('eth_getBlockByNumber', ['latest']))
+          .then((block) => block.timestamp * 2)
+          .then((timestampIncrease) => {
+            hardhat.send('evm_increaseTime', [hexValue(timestampIncrease)])
+            hardhat.send('hardhat_mine', [hexValue(1), hexValue(timestampIncrease)])
+          })
+
+        cy.then(() => hardhat.send('hardhat_mine', ['0x1', '0xc'])).then(() => {
+          cy.get(getTestSelector('dismiss-tx-confirmation')).click()
+
+          // ui check
```

Comment:
These comments should be more explicit, eg: `// Check that the balance has not changed`